examples/linkedlist.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import print_function, absolute_import
+from numba.utils import OrderedDict
+import numpy as np
+from numba import njit
+from numba.jitclass import jitclass
+from numba import int32, deferred_type, optional
+from numba.runtime import rtsys
+
+node_type = deferred_type()
+
+spec = OrderedDict()
+spec['data'] = int32
+spec['next'] = optional(node_type)
+
+
+@jitclass(spec)
+class LinkedNode(object):
+    def __init__(self, data, next):
+        self.data = data
+        self.next = next
+
+    def prepend(self, data):
+        return LinkedNode(data, self)
+
+
+@njit
+def make_linked_node(data):
+    return LinkedNode(data, None)
+
+
+node_type.define(LinkedNode.class_type.instance_type)
```

Comment:
> I think it could work simply as a local name for resolution when a jitclass is compiled.

The way I imagined the resolution would not be simple at all.  Can you elaborate?